Pull request/95d622ea

eschnett opened the pull request.

tkelman commented:
The FastArrays name is a bit questionable - it looks like the code was derived from your already-registered FlexibleArrays package, and in fact the tags that support 0.4 that you're registering here won't work because they used the old name. Why isn't it an in-place new major version for 0.5?

I'll merge the registration of SIMD separately, but would be good to rebase this once we've decided on the best course.

eschnett commented:
Tony

I renamed the package because I discovered at JuliaCon that the name
`FlexibleArrays` was ambiguous, and people didn't know what it means. My
original motivation for the package was to speed up indexing operations
(which it does), but I then unluckily named it `FlexibleArrays`.

While renaming it to `FastArrays` (suggestions for other names welcome), I
discovered that its implementation doesn't work in Julia 0.5 since one
cannot create new types in a function. Thus the implementation changed
completely -- this is now a rewrite from scratch rather than an update.
Since the new implementation doesn't work with Julia 0.4, I decided to
reinstate the old package `FlexibleArrays` and mark it 0.4 only, and keep
it around, pointing to `FastArrays` for Julia 0.5 and later.

Other solutions are possible, e.g. having both implementation in the same
package `FastArrays` and using `VERSION` to select a particular version.

The major features of `FastArrays` are:
- lower index bound can be anything, doesn't have to be 1
- array index bounds can be fixed (i.e. be part of the type), but don't
have to
- fixed bounds are faster (require fewer integer operations) for
multi-dimensional arrays

This is probably a superset of `ImmutableArrays`, or `OffsetArrays`, or
`FixedSizeArrays`.

Another name I have in mind is `SuperArrays`...

-erik



On Wed, Aug 17, 2016 at 3:09 PM, Tony Kelman <notifications@github.com>
wrote:

> The FastArrays name is a bit questionable - it looks like the code was
> derived from your already-registered FlexibleArrays package, and in fact
> the tags that support 0.4 that you're registering here won't work because
> they used the old name. Why isn't it an in-place new major version for 0.5?
>
> I'll merge the registration of SIMD separately, but would be good to
> rebase this once we've decided on the best course.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/METADATA.jl/pull/6005#issuecomment-240515448>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANCCjPMm0oSkxkY3CF8D3pGvoDppYJ5ks5qg1xxgaJpZM4JlsYe>
> .
>



-- 
Erik Schnetter <schnetter@gmail.com>
http://www.perimeterinstitute.ca/personal/eschnetter/

tkelman commented:
Okay. We should at least remove the tags that point to commits where the package was actually named FlexibleArrays, or modify the requires files of those tags here in METADATA so they can't be installed under Julia 0.4. Another option to clearly differentiate the new package from the old (if you do want to keep the same name, but that's not absolutely required) would be to add a Julia upper bound of 0.5 on the existing tags of the old FlexibleArrays implementation.
